### DISCUSSION TITLE
Simple project (sample) failed on VCL

### DISCUSSION THREAD

**Discussion by george-kar** [SchneiderInfosystems/FB4D#150] [Q&A] [answered]
It looks like that component is not fully compatible with VCL. I have copied the very basic components and functions from the FireMonkey example provided "RTDB_SimpleReadWrite"

Everything works getting and setting data to the database but when you close the form, I get the error

Project Project1.exe raised exception class ENetHTTPResponseException with message 'Error reading data: (12017) The operation has been canceled'.

IDE: Delphi 11.2 Alexandria

**Comment by SchneiderInfosystems** [comment]
When running the application in the Delphi IDE, you will see this silent exception of the class _ENetHTTPResponseException_ with message 'Error reading data: (12017) The operation has been canceled'. This is no defect because it is a silent exception which the user will never see.

Now I have added this explanation to the reference manual: https://github.com/SchneiderInfosystems/FB4D/wiki/FB4D-Reference-IFirestoreDatabase#note-about-the-exception-while-stopping-the-listener

**Comment by george-kar** [comment]
I disagree, as my application built with D11.2 returns an error in production too

```
---------------------------
Project1
---------------------------
Invalid pointer operation.
---------------------------
OK   
---------------------------

```

**Comment by george-kar** [comment]
Ok I found the issue, it looks like you forgot to add the "fEvent.StopListening" on the OnCloseQuery of the form.
I suggest to be added to all your sample projects when a listener is started and never stopped by the user.

**Comment by SchneiderInfosystems** [comment, marked as answer]
It should be stopped automatically within TFirebaseEvent.Destroy in this line:

```
  if fListener.IsRunning then
    StopListening;
```

Do you have declared the `fFirebaseEvent: IFirebaseEvent` properly as interface type? 
If you have declared fFirebaseEvent as TFirebaseEvent, you must release it yourself. This would explain your findings.
